stablehlo/reference/Ops.cpp

```diff
@@ -136,10 +140,18 @@ void failOnDecomposableOp(Operation &op) {
       op.getName().getStringRef().str().c_str()));
 }
 
+template <typename T>
+SmallVector<InterpreterValue> getInterpreterValues(SmallVector<T> list) {
+  SmallVector<InterpreterValue> result(list.size());
+  std::transform(list.begin(), list.end(), result.begin(),
+                 [](auto value) { return InterpreterValue(value); });
+  return result;
+}
```

Comment:
Can we push this complexity into Scope, and introduce similar overloads to what was done for `find` in the previous round of reviews?

Reply:
The only problem is now that there are three overloads,
```
void add(ValueRange ssaValues, ArrayRef<Tensor> runtimeValues);
void add(ValueRange ssaValues, ArrayRef<Token> runtimeValues);
void add(ValueRange ssaValues, ArrayRef<InterpreterValue> runtimeValues);
```
It will be ambiguous to call scope.add(op.getResults(), {result});

I've marked InterpreterValue constructor `explicit` to avoid ambiguity. Lmk if you have an alternative suggestion.

Reply:
Another option is to have bunch of overloads for `std::initializer_lists`, at which point the number of overloads in Scope seems excessive.

Reply:
Actually, I ended up doing neither. This version looks the best imo, ptal.

Reply:
This looks good, but instead of `.getResult(0)`, I think you should be able to just call `.getResult()`.

Reply:
You're right, not sure how I missed that one :)

Reply:
Nvm, turns out the idx is needed. I just remembered putting the 0 there because that function didn't exist.